tower/src/retry/future.rs

```diff
@@ -88,11 +88,8 @@ where
                     }
                 }
                 StateProj::Checking { checking } => {
-                    this.retry
-                        .as_mut()
-                        .project()
-                        .policy
-                        .set(ready!(checking.poll(cx)));
+                    ready!(checking.poll(cx));
```

Comment:
nit, take it or leave it: is this state really "checking" if we can retry?  it seems like if we are in this state, we have already decided that the request can be retried, but we are _waiting_ before actually sending the retry request --- perhaps this should be renamed to `State::Waiting`?

not a blocker for this branch, since that naming was there before, but it might be worth changing...

Reply:
Good point, updated.